src/settings/global_settings.rs

```diff
@@ -26,6 +28,129 @@ fn identifier_is_valid(id: &str) -> bool {
     id.len() <= 23 && id.chars().all(|x| x.is_alphanumeric() || x == '-')
 }
 
+#[derive(DeserializeFromStr, Copy, Clone, Debug)]
+pub struct IpAddr(pub smoltcp_nal::smoltcp::wire::Ipv4Address);
+
+impl Serialize for IpAddr {
+    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let mut display: String<16> = String::new();
+        write!(&mut display, "{}", self).unwrap();
+        serializer.serialize_str(&display)
+    }
+}
+
+impl IpAddr {
+    pub fn new(bytes: &[u8]) -> Self {
+        Self(smoltcp::wire::Ipv4Address::from_bytes(bytes))
+    }
+}
+
+impl core::str::FromStr for IpAddr {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let addr = smoltcp::wire::Ipv4Address::from_str(s).map_err(|_| "Invalid IP format")?;
+        Ok(Self(addr))
+    }
+}
+
+impl core::fmt::Display for IpAddr {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+#[derive(Tree, Clone, Debug)]
+pub struct Properties {
+    pub ip: IpAddr,
+    pub netmask: IpAddr,
+    pub gateway: IpAddr,
+    pub broker: IpAddr,
+    pub fan_speed: f32,
+    pub id: String<23>,
```

Comment:
This is a separate struct from BoosterMainBoardData so that we can hide away implementation details from miniconf (like the fact we have an `id_len` u32 and `id` is actually an array of bytes, etc.